src/main/java/io/castle/client/internal/backend/OkRestApiBackend.java

```diff
@@ -0,0 +1,230 @@
+        this.authenticate = baseUrl.resolve("/v1/authenticate");
+        this.reviewsBase = baseUrl.resolve("/v1/reviews/");
+        this.identify = baseUrl.resolve("/v1/identify");
+    }
+
+    @Override
+    public void sendTrackRequest(String event, String userId, JsonElement contextPayload, JsonElement propertiesPayload, final AsyncCallbackHandler<Boolean> asyncCallbackHandler) {
+        JsonObject json = new JsonObject();
+        json.add("name", new JsonPrimitive(event));
+        if (userId == null) {
+            json.add("user_id", JsonNull.INSTANCE);
+        } else {
+            json.add("user_id", new JsonPrimitive(userId));
+        }
+        json.add("context", contextPayload);
+        if (propertiesPayload != null) {
+            json.add("properties", propertiesPayload);
+        }
+        String content = json.toString();
+        RequestBody body = RequestBody.create(JSON, content);
+        Request request = new Request.Builder()
+                .url(track)
+                .post(body)
+                .build();
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                Castle.logger.error("HTTP layer. Error sending track request.", e);
+                if (asyncCallbackHandler != null) {
+                    asyncCallbackHandler.onException(e);
+                }
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                if (asyncCallbackHandler != null) {
+                    asyncCallbackHandler.onResponse(response.isSuccessful());
+                }
+            }
+        });
+    }
+
+    private Request buildAuthenticateRequest(String event, String userId, JsonElement contextPayload, JsonElement traitsPayload, JsonElement propertiesPayload) {
+        JsonObject json = new JsonObject();
+        json.add("name", new JsonPrimitive(event));
+        json.add("user_id", new JsonPrimitive(userId));
+        json.add("context", contextPayload);
+        if (propertiesPayload != null) {
+            json.add("properties", propertiesPayload);
+        }
+        if (traitsPayload != null) {
+            json.add("traits", traitsPayload);
+        }
+        RequestBody body = RequestBody.create(JSON, json.toString());
+        return new Request.Builder()
+                .url(authenticate)
+                .post(body)
+                .build();
+    }
+
+    @Override
+    public Verdict sendAuthenticateSync(String event, final String userId, JsonElement contextPayload, JsonElement propertiesPayload, JsonElement traitsPayload) {
+        Request request = buildAuthenticateRequest(event, userId, contextPayload, traitsPayload, propertiesPayload);
+        try {
+            Response response = client.newCall(request).execute();
+            return extractAuthenticationAction(response, userId);
```

Comment:
I think something is missing here. If I understand this correctly this would catch both 4XXand 5XX responses but should only 5XX see https://paper.dropbox.com/doc/Server-SDK-specs-5WNJdpiGefPk2I1aAtWET#:h2=Fail-over-Strategy:

Reply:
Good catch!.
There was a corner case here. If the authenticate response is has status 4xx or 5xx and the failover strategy is throw, then no exceptions was created.
I assume that the failover strategy should be used when I can not create a correct Verdict from the response, it means that:
- I expect a success status (code >= 200 && code < 300)
- The json is correct and parse to a Verdict object.
In all other cases I use the failover strategy.
If this should be different, just let me know.

Reply:
We should throw an error if it is in the 400 range otherwise we'll swallow tracking errors which we don't want to do. Failover strategy should only happen if something is wrong on our end 5XX codes but if there is an un-processable payload it should fail for the customer.

Reply:
Ok, I will fixit now.

Reply:
great!

Reply:
Done on commit 042ea4c